### PR TITLE
Added a section "Cloning the repository on Windows" for issue#1567

### PR DESCRIPTION
**Pull request by ankurshukla10**
This is for issue id #1567

### CODE REVIEW

**Review by marcingrzejszczak — APPROVED** [review]
Let's wait for the approval from our documentation guru @Buzzardo

File: docs/src/main/asciidoc/_building.adoc
Comment (Buzzardo) [inline docs/src/main/asciidoc/_building.adoc:5]: Change "*windows*" to "Windows" (no bold but capitalize it.

File: docs/src/main/asciidoc/_building.adoc
Comment (Buzzardo) [inline docs/src/main/asciidoc/_building.adoc:6]: Change to "result in an incorrectly (probably partially) checked out repository."

File: docs/src/main/asciidoc/_building.adoc
Comment (Buzzardo) [inline docs/src/main/asciidoc/_building.adoc:8]: Change to

To resolve this issue, you can set the `core.longPaths` attribute to `true` or clone the Spring Cloud Contract repository.
The following example shows how to set the `core.longPaths` attribute to `true`:

File: docs/src/main/asciidoc/_building.adoc
Comment (Buzzardo) [inline docs/src/main/asciidoc/_building.adoc:14]: Change to:

The following example shows how to clone Spring Cloud Contract:

**Review by Buzzardo — CHANGES_REQUESTED** [review]
I know that probably seems redundant, but it leads to maximum clarity, especially when people are skimming along to find an answer quickly (as is often the case). Thanks.

**Comment by ankurshukla10** [comment]
@Buzzardo  I have made the changes you requested. Please have a look.

File: docs/src/main/asciidoc/_building.adoc
Comment (Buzzardo) [inline docs/src/main/asciidoc/_building.adoc:17]: Remove this line. (Sorry. I missed it last time.)

**Review by Buzzardo — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/_building.adoc
Comment (ankurshukla10) [reply docs/src/main/asciidoc/_building.adoc:17]: @Buzzardo Sorry but its not really clear which line are you asking me to remove? Is this the "OR" at line#14.  ?

Comment (Buzzardo) [reply docs/src/main/asciidoc/_building.adoc:17]: Yes. Please remove the line that consists of `*_OR_*`.

Comment (ankurshukla10) [reply docs/src/main/asciidoc/_building.adoc]: @Buzzardo  I have removed that line and also removed a bullet point after it, for the sake of symmetry. Let me know what do you think.

File: docs/src/main/asciidoc/_building.adoc
Comment (Buzzardo) [reply docs/src/main/asciidoc/_building.adoc]: Since I am having an awful time seeing how the lines relate to one another, here's what the whole thing should be (note that there are no bullets anywhere):

------------------

While cloning this project on Windows, some files in the git repository may exceed the Windows maximum file path limit of 255 characters, which may result in an incorrectly (probably partially) checked out repository.

To resolve this issue, you can set the `core.longPaths attribute` to `true` or clone the Spring Cloud Contract repository.

The following example shows how to set the core.longPaths attribute to true:

```git config --system core.longpaths true```

The following example shows how to clone Spring Cloud Contract:

```git clone -c core.longPaths=true https://github.com/spring-cloud/spring-cloud-contract.git```

Comment (ankurshukla10) [reply docs/src/main/asciidoc/_building.adoc]: @Buzzardo It's now corrected.

File: docs/src/main/asciidoc/_building.adoc
Comment (ankurshukla10) [reply docs/src/main/asciidoc/_building.adoc:23]: @Buzzardo @shanman190 Sure! I like the idea of options myself. Will update this shortly.

Comment (shanman190) [reply docs/src/main/asciidoc/_building.adoc:23]: @ankurshukla10, let's wait for Buzzardo's opinion first. I'm kind of curious if we don't just take a simplified approach as what is typically done in git documentation and just use the system example only leaving it to the user to perform an alternative at their own preference.

Comment (Buzzardo) [reply docs/src/main/asciidoc/_building.adoc:23]: I generally favor more detail rather than less. However, this document isn't about how to configure Git, so we need to not get carried away. That said, I would explain that there are three options and make them into an unnumbered (bulleted) list, thus:

To set the `core.longPaths` attribute to `true`, you have three options:

- Change it for all users of the machine (doing so requires administrator privileges):
+
====
[source,bash]
----
git config --system core.longPaths true
git clone https://github.com/spring-cloud/spring-cloud-contract.git
----
====
- Change it for the current user (no administrative privileges required):
+
====
[source,bash]
----
git config --global core.longPaths true
git clone https://github.com/spring-cloud/spring-cloud-contract.git
----
====

- Change for just this repository (administrative privileges depend on where the repository is being cloned to):
+
====
[source,bash]
----
git clone -c core.longPaths true https://github.com/spring-cloud/spring-cloud-contract.git
----
====

Comment (Buzzardo) [reply docs/src/main/asciidoc/_building.adoc:23]: It turned the code fences into headings. Sorry about that.

Comment (ankurshukla10) [reply docs/src/main/asciidoc/_building.adoc:32]: @Buzzardo @shanman190  The changes are made, please review and let me know.

Comment (Buzzardo) [reply docs/src/main/asciidoc/_building.adoc:32]: Please add the code fences. I know it seems like a lot of structure for not much benefit, but it makes all the code listings be consistent. If we use the three backticks approach, these code listings look different than our other code listings. (This all happens because some code listings have callouts, and we have to add this structure to make the code listings with callouts look the same as the listings that don't have callouts. Sorry about the extra work.

Comment (ankurshukla10) [reply docs/src/main/asciidoc/_building.adoc:32]: @Buzzardo 
Just to understand it correctly, am I supposed to remove the back-ticks and use the [source,bash], so now this becomes

[source,bash]
git config --system core.longPaths true
git clone https://github.com/spring-cloud/spring-cloud-contract.git

Comment (Buzzardo) [reply docs/src/main/asciidoc/_building.adoc:32]: Github formatting letting us down. Let's see if I can make a code block to show the Asciidoc structure:

```
====
[source,bash]
----
git config --system core.longPaths true
git clone https://github.com/spring-cloud/spring-cloud-contract.git
----
====
```

And then do the same for each of the three. Again, I know it's a lot of format for not much code, but it does solve our consistency problem so that all the code listings look the same. Thanks.

Comment (ankurshukla10) [reply docs/src/main/asciidoc/_building.adoc:35]: @Buzzardo  Thanks, made those changes.

**Review by Buzzardo — APPROVED** [review]

File: docs/src/main/asciidoc/_building.adoc
Comment (Buzzardo) [reply docs/src/main/asciidoc/_building.adoc:35]: I have approved the changes I requested. Thanks much.

**Pull request merged**
